google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/CloudDebugSettingsEditor.java

```diff
@@ -47,7 +60,21 @@ protected JComponent createEditor() {
   }
 
   @Override
-  protected void resetEditorFrom(CloudDebugRunConfiguration runConfiguration) {
-    settingsPanel.setProjectName(runConfiguration.getCloudProjectName());
+  protected void resetEditorFrom(@NotNull CloudDebugRunConfiguration runConfiguration) {
+    String projectId = runConfiguration.getCloudProjectName();
```

Comment:
Previous code was setting *just project ID* (disguised as project name again in code) and never set user name (email) in `CloudDebugRunConfiguration`. As far as I understand, previous ProjectSelector would load all projects anyway and would search for any project of any user with matching ID? We don't do it anymore and load on demand only, plus match projects strictly per user account, should we add user name into `CloudDebugRunConfiguration` as well?

Reply:
are you basically asking if IDs are unique across users? if so, the answer is yes. project IDs are unique across all of GCP as far as I understand it

Reply:
Yes, there are a few places where both ID and user name is stored, and here only project ID is stored. It looks like the code assumed it is enough to find project match later, but this needs loading of all projects for all available accounts. Do you think it makes sense to add project name and user name in cloud debug configuration as well to avoid re-loading all projects?

Reply:
Previously, it would have to reload all the projects too right? If so I would avoid any optimization here to keep the PR concise (we could do it separately if we wanted).

Reply:
Also any updates to the persisted configuration requires consideration of backwards compatibility with already stored configs. I definitely wouldn't want to go down that path in this PR since its meant to just be updating the project selector to the new widgets.

Reply:
Sure, there are no changes or optimizations in this PR, we can still use the same project ID it already holds, it just affects that project selector won't show the project name.